src/cigar.c

```diff
@@ -8,12 +8,46 @@ void edits_to_cigar(const char *edits, char **cigar)
 {
     size_t n = strlen(edits);
     *cigar = malloc(2 * n + 1); // maximal length for a cigar
+
+
     // Compute the CIGAR string
+    memset(*cigar, 0,2*n+1);
+
+    char cur = edits[0];
+    int no = 1;
+
+    for (int i = 1 ; i <= n ; i++) {
+        if (edits[i] == cur) {
+            no++;
+        }
+        else {
+            char *num = no + '0';
+            strncat(*cigar, &num, 1);
+            strncat(*cigar, &cur, 1);
+            cur = edits[i];
+            no = 1;
+        }
+    }
 }
 
 static inline int count_edits(const char *cigar)
 {
-    return 42; // Work out how long the string will be
+    int n = strlen(cigar);
+    int cum = 0;
+    int len = 0;
+    for (int i = 0 ; i < n ; i++ ) {
+        if (cigar[i] == '0') {
+            if (atoi(&cigar[i]) != 0 ) {
+                len += atoi(&cigar[i]);
+                if (atoi(&cigar[i-1]) != 0 ) {
+                    len += atoi(&cigar[i-1])*9;
+                }
+            }
+            else continue;
```

Comment:
Isn't this `else continue` the same as nothing? You are not doing anything more in this loop iteration if you end up here.